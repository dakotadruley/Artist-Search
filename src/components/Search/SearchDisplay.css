.SearchDisplay {{}

    #search {
        display: inline-block;
        margin: 1rem 0;
        padding: 1rem;
    }

    button {
        display: inline-block;
        margin: 1rem;
    }
}