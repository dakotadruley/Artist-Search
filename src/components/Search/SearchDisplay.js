import React from 'react';
import PropTypes from 'prop-types';

const SearchDisplay = ({ name, onChange, onClick }) => (
  <section>
    <input id="search" type="text" name={name} value={name} placeholder="Enter Artist Name..." onChange={onChange}/>
    <button onClick={onClick}>Search</button>
  </section>
);

SearchDisplay.propTypes = {
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
};

export default SearchDisplay;

