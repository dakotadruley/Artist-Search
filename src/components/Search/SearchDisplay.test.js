import React from 'react';
import { shallow } from 'enzyme';
import SearchDisplay from './SearchDisplay.js';

describe('SearchDisplay', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<SearchDisplay
      name={'dakota superstar'}
      onChange={() => {}}
      onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
