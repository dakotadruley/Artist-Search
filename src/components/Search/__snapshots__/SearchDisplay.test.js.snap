// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDisplay matches a snapshot 1`] = `
<section
  className="SearchDisplay"
>
  <input
    id="search"
    name="dakota superstar"
    onChange={[Function]}
    placeholder="Enter Artist Name..."
    type="text"
    value="dakota superstar"
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</section>
`;
